test/modules/OS/POSIX/select.chpl

```diff
@@ -0,0 +1,125 @@
+use OS.POSIX;
+
+// This test works by writing to and reading from, in separate tasks,
+// many pipes simultaneously.  The select_posix() calls are used to
+// decide which pipe(s) to operate upon, based on whether we have
+// space for writing and/or data available for reading.
+
+config const numPipes = 2;
+config const numXfers = numPipes * 4 + 1;
+
+var fdsRead: [0..<numPipes] c_int;
+var fdsWrite: [0..<numPipes] c_int;
+
+for i in 0..<numPipes {
+  var fildes:2*c_int;
+  writeln(pipe(c_ptrTo(fildes)) == 0);
+  fdsRead[i] = fildes(0);
+  fdsWrite[i] = fildes(1);
+}
+
+cobegin {
+  beReader(fdsRead);
```

Comment:
same issue as the pipes test - if there was only 1 thread, we would have a problem here, right?

Reply:
This is addressed by https://github.com/chapel-lang/chapel/pull/19632/commits/eabbd841b89137ea9b7ade4655241a50e85b5bc7.